Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -76,44 +76,49 @@
 	% (that would generate a mean control) is skipped and then we use the decoding along columns, goind across
 	% rows generates different decoded volumes.
 	% #### For Walsh Decoding Matrix ####
-	if strcmp(xQ.TimeEncodedMatrixType,'Walsh')
+	if strcmpi(xQ.TimeEncodedMatrixType,'Walsh')
 		
 		if xQ.TimeEncodedMatrixSize == 4
 			xQ.TimeEncodedMatrix =...
-				[1 -1  1 -1;
+				[1  1  1  1; 
+				 1 -1  1 -1;
 				 1 -1 -1  1;
 				 1  1 -1 -1];
 		elseif xQ.TimeEncodedMatrixSize == 8
-            xQ.TimeEncodedMatrix = [1 -1  1 -1  1 -1  1 -1;
-                                 1 -1  1 -1 -1  1 -1  1;
-                                 1 -1 -1  1 -1  1  1 -1;
-                                 1 -1 -1  1  1 -1 -1  1;
-                                 1  1 -1 -1  1  1 -1 -1;
-                                 1  1 -1 -1 -1 -1  1  1;
-                                 1  1  1  1 -1 -1 -1 -1];
+            xQ.TimeEncodedMatrix = [1  1  1  1  1  1  1  1;
+				                    1 -1  1 -1  1 -1  1 -1;
+                                    1 -1  1 -1 -1  1 -1  1;
+                                    1 -1 -1  1 -1  1  1 -1;
+                                    1 -1 -1  1  1 -1 -1  1;
+                                    1  1 -1 -1  1  1 -1 -1;
+                                    1  1 -1 -1 -1 -1  1  1;
+                                    1  1  1  1 -1 -1 -1 -1];
 		end
 		
 		% #### For Hadamard Decoding Matrix ####
-	elseif strcmp(xQ.TimeEncodedMatrixType,'Hadamard')
+	elseif strcmpi(xQ.TimeEncodedMatrixType,'Hadamard')
+		% An alternative Philips version xQ.TimeEncodedMatrix = [1,1,1,1;1,1,-1,-1;1,-1,1,-1;1,-1,-1,1];
```

Comment:
I would put this here in the same format as the others (so not 1 continuous row but also a square)

Reply:
Agree, will do.

Reply:
Check this link https://github.com/ExploreASL/ExploreASL/pull/1556/commit/63feb0179603254c0b740de50f64244c6baccbb8

Reply:
Two those things are fixed [here](https://github.com/ExploreASL/ExploreASL/pull/1556/commits/126a289a058a64b290bdbd4d59f4dacbb38d9e19)